add documentation for Azure Monitor Pod Identity (#227)
kedacore/keda-docs: content/docs/2.0/scalers/azure-monitor.md

    metricFilter: namespace eq 'default'
    metricAggregationInterval: "0:1:0"
    targetValue: "1"

>     activeDirectoryClientId: CLIENT_ID_ENV_NAME # Optional if podIdentity is defined

Reviewer: Let's be more precise:
```suggestion
    activeDirectoryClientId: CLIENT_ID_ENV_NAME # Optional, can use TriggerAuthentication as well
```

>     activeDirectoryClientPassword: CLIENT_PASSWORD_ENV_NAME # Optional if podIdentity is defined

Reviewer: Let's be more precise:

```suggestion
    activeDirectoryClientPassword: CLIENT_PASSWORD_ENV_NAME # Optional, can use TriggerAuthentication as well
```

### Authentication Parameters

You can use `TriggerAuthentication` CRD to configure the authentication by providing a set of Azure Active Directory credentials

> or by using pod identity.

Reviewer: Please move this on the line above

Author: done

    - parameter: activeDirectoryClientPassword
      name: azure-monitor-secrets
      key: activeDirectoryClientPassword
  # or Pod Identity, kind: Secret is not required in case of pod Identity
  podIdentity:

>       provider: azure    

Reviewer: ```suggestion
      provider: azure
```
